### PR TITLE
feat: track enable notifications impression

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-904 #done

[WT-904]: https://dailydotdev.atlassian.net/browse/WT-904?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sshanzel** [comment]
@rebelchris I have added the new impression tracking for the banner. I'm not sure if this is the only required change as I didn't notice any change in the other events.

File: packages/shared/src/components/notifications/EnableNotification.tsx [hunk 1]
```diff
@@ -69,19 +69,34 @@ function EnableNotification({
...
+  const shouldNotDisplay = conditions.some((passed) => passed);
+
+  useEffect(() => {
+    if (shouldNotDisplay) {
+      return;
+    }
+
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_id: source,
```
Comment (idoshamun) [inline packages/shared/src/components/notifications/EnableNotification.tsx:88]: Can you please add the target id to the click event as well please?

Comment (sshanzel) [reply packages/shared/src/components/notifications/EnableNotification.tsx:88]: On click event I believe it was also included but was applied on the `origin` property. Should I move the placements below:
https://github.com/dailydotdev/apps/blob/3a66e6938c1d3e7478ffb083a08c2d938308e893/packages/shared/src/hooks/useEnableNotification.tsx#L29-L32

Comment (idoshamun) [reply packages/shared/src/components/notifications/EnableNotification.tsx:88]: Oh then let's do the same for the impression

Comment (sshanzel) [reply packages/shared/src/components/notifications/EnableNotification.tsx:88]: Cool. I will send them a message too, just so they will be aware.

Comment (idoshamun) [reply packages/shared/src/components/notifications/EnableNotification.tsx:88]: Let me know when you change it so I can approve

**Review by idoshamun — APPROVED** [review]

**Review by rebelchris — APPROVED** [review]
Looks good, please don't forget to update the taxonomy sheet with this change and which ones are already in place (column F)

**Comment by sshanzel** [comment]
> Looks good, please don't forget to update the taxonomy sheet with this change and which ones are already in place (column F)

Done! Thanks for letting me know that I can actually update it too.

**Pull request merged**